### DISCUSSION TITLE
msys2/MSYS2-packages-contrib?

### DISCUSSION THREAD

**Discussion by ghost** [msys2/MSYS2-packages#4405] [Ideas]
You said you don't want to divert too much from Cygwin. But the needs for new packages can't be ignored. A lot of new packages are submitted as PRs. What about making a new repository for these packages? MSYS2 out of the box will not have the repository `msys-contrib` enabled. Users want to use it have to enable it manually via editing `pacman` config file.

**Comment by Biswa96** [comment]
You can create your own pacman repository and share it with others. It does not have to be in msys2.

**Reply by ghost** [reply]
Imagine each people here created their own pacman repositories: https://github.com/msys2/MSYS2-packages/pulls?q=is%3Apr+is%3Aopen+label%3Anew-package

**Reply by sskras** [reply]
If you follow @Biswa96 scenario, it would be effective to ask authors of the PRs to change their `origin` to the new (non-official) repo OR just recreate [a new PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork#:~:text=In%20the%20%22base%20branch%22%20dropdown%20menu%2C%20select%20the%20branch%20of%20the%20upstream%20repository%20you%27d%20like%20to%20merge%20changes%20into.):

![](https://docs.github.com/assets/cb-96554/mw-1440/images/help/pull_requests/choose-base-fork-and-branch.webp)

If the new repo, e.g. `iahung2/msys-contrib` is a fork of `msys2/MSYS2-packages`, it will appear in the **base repository** list.

---

Eg. my fork (of an unrelated project) can see other forks:

![image](https://github.com/msys2/MSYS2-packages/assets/7887758/b989e38b-43db-4bb9-aff9-fc560786cb70)

... and I am able to recreate PR for a different (but forked) repo:

![image](https://github.com/msys2/MSYS2-packages/assets/7887758/c6aa3a6e-ee17-445e-a38c-058f8e4cd84d)

---

@iahung2, do you have personal resources to maintain `msys-contrib` ? :)

Of course. a separate organization can be created to manage the project easier, so the final repo name becomes something like `msys2-contrib/msys-contrib`.

**Reply by ghost** [reply]
> @iahung2, do you have personal resources to maintain `msys-contrib` ? :)

No. This is the reason why I think something like `msys-contrib` should be under the umbrella of the MSYS2 project.

**Reply by sskras** [reply]
This would go against the [original project aims](https://www.msys2.org/wiki/How-does-MSYS2-differ-from-Cygwin/#:~:text=MSYS2%20provides%20a%20minimal%20shell%20required%20to%20run%20autotools%20and%20other%20build%20systems&text=configure%20them%20and%20build%20them&text=allow%20porting%20Unix%20programs,without%20requiring%20a%20POSIX%20emulation%20layer&text=so%20the%20number%20of%20provided%20POSIX%2Demulated%20software%20is%20very%20small):

![image](https://github.com/msys2/MSYS2-packages/assets/7887758/b82a969d-8a34-4707-8ace-f5de6a185a4f)

**Reply by sskras** [reply]
It's a bit sad to see that the discussion was marked a resolved.

@iahung2, I very much support your aim and I would like to contribute to such 3rd party repo. :)
I just haven't developed the needed skills yet.

Although I have made some steps in researching experiments in that area:
https://disconnected.systems/blog/archlinux-repo-in-a-git-repo/
https://github.com/yujinakayama/pacman-repo
https://github.com/KSXGitHub/pacman-repo

There are even tutorials with `*.github.io` FQDNs in use:
https://www.sainnhe.dev/post/create-personal-arch-linux-package-repository/
https://www.arcolinuxiso.com/how-to-create-your-own-online-arch-linux-repository-on-github-and-use-it-on-any-arcolinux/

**Reply by ghost** [reply]
Neither you or me will have the ability to maintain such a repo in the long term.

**Reply by sskras** [reply]
Neither have the current contributors / maintainers, I think.
That's why it's important to gather an additional community. :P

**Comment by sskras** [comment]
> But the needs for new packages can't be ignored. A lot of new packages are submitted as PRs.

Another option would be learning to use the Midipix project: https://midipix.org/#sec-midipix

... and research porting packages there, e,g.: https://dev.midipix.org/build/midipix_build/c/1d6d564?branch=main
